From cd417f16317b47958e9ce8e2e504a358200b2433 Mon Sep 17 00:00:00 2001
From: Joshua Harlan Lifton <lifton@puppetlabs.com>
Date: Mon, 12 Mar 2012 17:52:38 -0700
Subject: [PATCH] Add account information option to header

Previous to this commit, Dashboard did not have any way to display
user account information. With this commit, an account_widgets
callback is added to the global navigation header.
---
 app/views/shared/_global_nav.html.haml |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/app/views/shared/_global_nav.html.haml b/app/views/shared/_global_nav.html.haml
index 4be7116..b9dbd98 100644
--- a/app/views/shared/_global_nav.html.haml
+++ b/app/views/shared/_global_nav.html.haml
@@ -1,13 +1,16 @@
 .topbar
   .topbar-inner
     .container
-      %ul#global-navigation.navigation
+      %ul#logo-and-account.navigation
         %li#logo
           - css = ["background-repeat: no-repeat", "text-indent: -9000px"]
           - css << "background-image: url('#{SETTINGS.custom_logo_url}')"
           - css << "height: #{SETTINGS.custom_logo_height}"
           - css << "width: #{SETTINGS.custom_logo_width}"
           %a{:href => root_path, :title => SETTINGS.custom_logo_alt_text, :style => css.join('; ')} Puppet Dashboard
+        - Registry.each_callback :core, :account_widgets do |callback|
+          = callback.call self
+      %ul#global-navigation.navigation
         %li#dashboard-version{:class => active_if(request.url == release_notes_url)}
           = link_to h(APP_VERSION), APP_VERSION_LINK
         %li#navigation-home{:class => active_if(request.url == root_url)}
-- 
1.7.5.4

